### PR TITLE
fix library ref and casts in tlsio_mbedtls

### PR DESCRIPTION
**Pull request by ewertons**
This updates and replaces https://github.com/Azure/azure-c-shared-utility/pull/447

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -535,10 +535,10 @@ endif()
 
 if(${use_mbedtls})
     if (WIN32)
-        set(aziotsharedutil_target_libs ${aziotsharedutil_target_libs} 
-            $ENV{mbedtlsLibDir}/mbedtls 
-            $ENV{mbedtlsLibDir}/mbedx509 
-            $ENV{mbedtlsLibDir}/mbedcrypto
+        set(aziotsharedutil_target_libs ${aziotsharedutil_target_libs} ws2_32
```
Comment (danewalton) [inline CMakeLists.txt:538]: what is `ws2_32`?

File: adapters/platform_win32.c [hunk 2]
```diff
@@ -20,6 +20,9 @@
 #if USE_WOLFSSL
 #include "azure_c_shared_utility/tlsio_wolfssl.h"
 #endif
+#if USE_MBEDTLS
```
Comment (danewalton) [inline adapters/platform_win32.c:23]: ideally should these all be `#if` and `#elif`?

File: CMakeLists.txt (hunk 1 above)
Comment (ewertons) [reply CMakeLists.txt:538]: Windows socket library.
https://docs.microsoft.com/en-us/windows/win32/winsock/creating-a-basic-winsock-application

File: adapters/platform_win32.c (hunk 2 above)
Comment (ewertons) [reply adapters/platform_win32.c:23]: We could. But the effect would be similar anyway if only one define is provided. 
On the other side, if more than one is provided, we better try to include everything and let the compilation fail, so customers know right away what is wrong.

**Pull request merged**